libr/bin/bin.c

```diff
@@ -32,17 +32,9 @@ static RBinPlugin *bin_static_plugins[] = { R_BIN_STATIC_PLUGINS, NULL };
 static RBinXtrPlugin *bin_xtr_static_plugins[] = { R_BIN_XTR_STATIC_PLUGINS, NULL };
 static RBinLdrPlugin *bin_ldr_static_plugins[] = { R_BIN_LDR_STATIC_PLUGINS, NULL };
 
-// TODO: try to deprecate and just call the r_bin_load_io_at_offset_as_az
-R_API bool r_bin_load_io (RBin *bin, int fd, ut64 baseaddr, ut64 loadaddr, int xtr_idx, ut64 offset, const char *name) {
-	// adding file_sz to help reduce the performance impact on the system
-	// in this case the number of bytes read will be limited to 2MB
-	// (MIN_LOAD_SIZE)
-	// if it fails, the whole file is loaded.
-	const ut64 MAX_LOAD_SIZE = 0;  // 0xfffff; //128 * (1 << 10 << 10);
-	int res = r_bin_load_io2 (bin, fd, baseaddr,
-		loadaddr, xtr_idx, offset, name, MAX_LOAD_SIZE);
-	return res? res: r_bin_load_io2 (bin, fd, baseaddr,
-			loadaddr, xtr_idx, offset, name, UT64_MAX);
+R_API bool r_bin_load_io(RBin *bin, int fd, ut64 baseaddr, ut64 loadaddr, int xtr_idx, ut64 offset, const char *name) {
+	r_return_val_if_fail (bin, false);
+	return r_bin_load_io2 (bin, fd, baseaddr, loadaddr, xtr_idx, offset, name, 0);
```

Comment:
the second call to r_bin_load_io2 has been removed because of what's explained in https://github.com/radare/radare2/pull/11868/commits/5a151c0782e574328cc7ab7f4771415a847e9d87